File: bin/crocetcions_ohrgrauqitophes.r
#!/usr/bin/rebol -wqs
rebol []
;"""Un programme mettant en évidence l'inutilité de l'orthographe pour la compréhension d'un texte"""
;"""Un proarmgme mtatent en éeicnvde ltii'ltniué de lghportr'aohe pour la crohepénosimn du'n txete """

; d'abord écrit en python, traduit en rebol

random/seed now/time/precise

ligne: does [ loop 80 [prin "_"] print "" ]

; début du programme
cmd_line: system/script/args
; si rien n'est spécifié en argument, on met une phrase lambda; 
; sinon, on traite tout le reste de la ligne de commande:
either (cmd_line) [
	text: cmd_line
	][
	text: "Arrêtez de m'emmerder avec les corrections orthographiques de merde"
	]

ligne
print "               .oO== texte original:   ==Oo."
print text
ligne
print ""
ligne
print "               .oO== txete oirignal:   ==Oo."

mots: parse text " "
textout: copy ""
foreach mot mots [
	either ((length? mot) > 3) [	; mot de plus de 3 lettres: interversion aléatoire
		motout: copy ""				; mot en sortie, qui sera invretrtvi
		append motout first mot
		append motout (random (copy/part at mot 2 ((length? mot) - 2) ) )
		append motout last mot
		append textout join motout " "
	][
		append textout join mot " "
	]
]

print textout
ligne

File: bin/crocetcions_ohrgrauqitophes.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""Un programme mettant en evidence l'inutilite de l'orthographe pour la comprehension d'un texte"""
from random import *
import string, sys

def scramble_text(text):
    mots = text.split(" ")
    textout = []
    mot = ""
    for mot in mots:
        mott = []                                      #liste equivalent au mot
        motout = []                                    #mot en sortie, qui sera invretrtvi
        for i in range(len(mot)): mott.append(mot[i])  #on convertit simplement le mot en une liste de lettres; pour pouvoir popper, plus loin
        if len(mot)>3:                                 #sinon, un mot de 3 lettres ne change pas
            motout.append(mot[0])
            fin=mott.pop()
            while (len(mott)>1):                      
                #for j in range(len(mott)):
                motout.append(mott.pop(randrange(1, (len(mott)))))
            motout.append(fin)
            textout.append(motout)
        else:
            textout.append(mott)
        textout.append(" ")
    # FIXME boucler si jamais le mot mélangé est égal au mot initial
    # FIXME si la fin du mot est une (ou plusieurs) ponctuation(s), la stocker, réduire le mot, traiter le mot, puis rajouter la(les) ponctuation(s)

    textsortie = ""
    for tmp in textout:
            word = ""
            for ltr in tmp:
                    word = word+ltr #+" "
            textsortie = textsortie+word
    
    return(textsortie)



if (__name__ == '__main__'):
    cmd_line = (sys.argv)
    if len(cmd_line) > 1:
        text = cmd_line[1]
    else:
        text ="Arrêtez de m'emmerder avec les corrections orthographiques de merde"

#print('_' * 80)
#print('               .oO== texte original:   ==Oo.')
#print(text)
#print('_' * 80)
#print('')
#print('_' * 80)
#print('               .oO== txete oirignal:   ==Oo.')
print(scramble_text(text))
#print('_' * 80)

